templates/etc/ansible/facts.d/checkmk_agent.fact.j2

```diff
@@ -1,9 +1,17 @@
-{% set checkmk_agent__tpl_plugin_list = checkmk_agent__plugin_list %}
-{% if checkmk_agent__plugin_autodetect|d() %}
-{%   for hostgroup in hostvars[inventory_hostname]['group_names'] | intersect(checkmk_agent__group_plugin_map.keys()) %}
-{%     set _ = checkmk_agent__tpl_plugin_list.append(checkmk_agent__group_plugin_map[hostgroup]) %}
+{% import 'templates/import/debops__tpl_macros.j2' as debops__tpl_macros with context %}
+{% set checkmk_agent__tpl_plugins = checkmk_agent__plugins + checkmk_agent__group_plugins + checkmk_agent__host_plugins %}
+{% if checkmk_agent__plugin_autodetect|d() | bool %}
+{%   set _ = checkmk_agent__tpl_plugins.append(checkmk_agent__autodetected_plugins) %}
+{%   for fact_name in ansible_local.keys() | intersect(checkmk_agent__facts_plugin_map.keys()) %}
+{%     if fact_name in ["mariadb", "mysql", "postgresql"] %}
+{%       if ansible_local[fact_name].server|d() in [ checkmk_agent__fqdn, "localhost" ] %}
+{%         set _ = checkmk_agent__tpl_plugins.append(checkmk_agent__facts_plugin_map[fact_name]) %}
+{%       endif %}
+{%     else %}
+{%       set _ = checkmk_agent__tpl_plugins.append(checkmk_agent__facts_plugin_map[fact_name]) %}
+{%     endif %}
 {%   endfor %}
 {% endif %}
-{
-"checkmk_agent__plugin_list" : {{ checkmk_agent__tpl_plugin_list | to_nice_json }}
-}
+{{ ({
+    "plugins": (debops__tpl_macros.flattened(checkmk_agent__tpl_plugins) | from_json | unique),
```

Comment:
Ok, something I forgot to mention: In my opinion it would make sense to introduce a page e.g. `docs/facts.rst`, where the format of the facts is described. As I started to discuss before in https://github.com/debops/debops-policy/issues/16#issuecomment-260730683 it would be nice to have a clear and mostly stable reference about the individual role facts so that a any 3rd-party tool evaluating the facts (e.g. asset inventory tool or such) can reason about the configuration of the role without forcing the author to read and understand the facts template. What do you think?